src/shared/codelens/codeLensUtils.ts

```diff
@@ -169,7 +189,11 @@ export async function pickAddSamDebugConfiguration(
         if (!templateItem) {
             return undefined
         }
-        await addSamDebugConfiguration(templateItem, TEMPLATE_TARGET_TYPE)
+        if (val.label.endsWith(WITH_API_CONFIG)) {
```

Comment:
Since you have access to the `templateItem`, you should have access to the `hasApiEventSource` value...that should be more definitive than a string match.

Reply:
Good call, changed conditional.